### DISCUSSION TITLE
More flexy integration with custom build process

### DISCUSSION THREAD

**Discussion by KrzysztofZawisla** [withastro/roadmap#433] [Proposal]
Hi,

I have searched for solutions that allow to integrate .astro files with custom build processes. I would like to bundle astro files into a webpack bundle. It's kinda required for me to omit crazy solutions with step building. I guess it's possible by accessing internal tools in astro core folder and porting it from the vite to the webpack. Anyone had similar idea and made first steps or deeper research about it?

**Comment by KrzysztofZawisla** [comment]
That's the project in which we would like to integrate astro in a custom way

https://github.com/ZSBRybnik/ZSB
https://github.com/ZSBRybnik/frontend/tree/master
https://github.com/ZSBRybnik/backend/tree/master

**Comment by florian-lefebvre** [comment]
Vite is an essential piece of Astro so it will not be swappable. However, you may want to use the Container API: https://docs.astro.build/en/reference/container-reference/
